content/c++/encyclopedia/exceptions-in-cpp.md

```diff
@@ -0,0 +1,133 @@
+---
+Title: "Exceptions in C++"
+Subjects:
+  - "Computer Science"
+  - "Game Development"
+Tags:
+  - "Catch"
+  - "Errors"
+  - "Try"
+Catalog Content:
+  - "https://www.codecademy.com/learn/learn-c-plus-plus"
+  - "https://www.codecademy.com/learn/paths/computer-science"
+  - "https://github.com/Codecademy-Curriculum/codepedia-content/blob/main/content/language-agnostic/encyclopedia/what-is-an-error.md"
+---
+
+## Catching an Exception
+
+The function `getString` as defined below will throw an error if you try to access an index outside the allowable bounds.
+
+```codebyte/cpp
+std::string getString(std::string values[], int index) {
+    return values[index];
+}
+
+int main()
+{
+    std::string words[] = { "one", "two", "three" };
+    
+    // this is fine, and will send one to the buffer
+    std::cout << getString(words, 0);
+
+    // this is not fine, and will most likely throw a std::length_error
+    std::cout << getString(words, 3);
+
+    return 0;
+}
+```
+
+Errors, when uncaught, will cause your program to immediately stop. This behavior can be prevented by wrapping the code which might emit an error in a `try` block, and providing at least one `catch` block to execute if the code throws the error. Look at the improvement below:
+
+```codebyte/cpp
+std::string getString(std::string values[], int index) {
+    return values[index];
+}
+
+int main()
+{
+    std::string words[] = { "one", "two", "three" };
+    
+    try {
+      // "one" will be sent to the buffer
+      std::cout << getString(words, 0);
+
+      // this line will throw an error and move immediately move program execution to the catch block
+      std::cout << getString(words, 3);
+    } 
```

Comment:
Let's use 4 here, just in case some learners might not realize that the `std::string` is zero-indexed. Don't want to test learners on something unrelated in this example.